m3/reporter.go

```diff
@@ -427,6 +427,14 @@ func (r *reporter) reportCopyMetric(
 		copy.MetricValue.Timer = t
 	}
 
+	// NB(r): This is to avoid sending on a closed channel,
+	// it's faster to actually defer/recover than acquire
+	// a read lock here to ensure we aren't closed: benchmarked
+	// this with BenchmarkTimer in reporter_benchmark_test.go
+	defer func() {
```

Comment:
It's roughly 350ns vs 630ns.  That's a pretty big price considering that timers are very high QPS.